### ISSUE TITLE
bug: `stream_maps` configs inside environments arent used

### ISSUE THREAD

**Issue by pnadolny13** [meltano/meltano#6942]
### Meltano Version

2.8.0

### Python Version

3.9

### Bug scope

Configuration (settings parsing, validation, etc.)

### Operating System

Mac

### Description

When I configure `stream_maps` at the top level for my loader (target-apprise) it works as expected and when I run `meltano config target-apprise` I get:

```
{
  "uris": [
    "${SLACK_WEBHOOK_URI}"
  ],
  "stream_maps": {
    "my_stream": {
      "title": "TITLE",
      "body": "BODY"
    }
  }
}
```

but when I put the config inside an environment I run the same `meltano config target-apprise` and I only get:

```
{
  "uris": [
    "${SLACK_WEBHOOK_URI}"
  ]
}
```

which doesnt include my stream_map config that I need to set at the environment level.

**Comment by aaronsteers** [comment]
@pnadolny13 - Thanks for raising. Mappers are not treated the same at other plugins so I can see how this could be an issue.

In terms of priority, do you have a workaround as of now?

**Comment by pnadolny13** [comment]
@aaronsteers to clarify this is not a mapper plugin this is the mapping config associated with an SDK based target, in this case the target is SDK based so I chose to do mapping there. I can give a standalone mapper a try also. The work around is to pull those configs out to env vars which works for me but is a bit painful especially for new users and knowing thats an option.

I wanted to share my user experience here in case the feedback is helpful 😄  - the environments and env var templating is so close to being clean and nice to use but it still has sharp edges like this. I was flying implementing tap-snowflake -> target-apprise, got it working (wow! literally 15 mins), then when I went to move it into my environments and pull the sensitive URI out to an env var because I wanted to open a PR I hit a bunch of walls, array env vars dont get rendered and the stream map disappeared when moved to my environment. This was a case where it was delightfully/impressively easy to get it working but hard to get it over the finish line and into prod. I probably spent 15-20 mins on getting it to work which was incredible but then 30-40 mins debugging and ultimately figuring out what was going on.

cc @tayloramurphy 

And to be clear I'm not trying to push this to be prioritized I'm only sharing my experience so you all can take it into consideration and decide if its worth prioritizing.

**Comment by edgarrmondragon** [comment]
@pnadolny13 yeah, https://github.com/meltano/meltano/issues/3171 keeps coming back...

Thanks for digging into the underlying issue 👍

**Comment by aaronsteers** [comment]
@edgarrmondragon and @pnadolny13 - The issue description of #3171 refers to arrays - this issue relates to subnodes of objects, but it also introduces the distinction of object keys versus values.


Do we think the issue is with one, some, or possibly all of these?:

1. items in config arrays
2. subnodes of objects
3. keys vs values

**Comment by pnadolny13** [comment]
@aaronsteers yeah youre right, its slightly different than described in https://github.com/meltano/meltano/issues/3171. My understanding is that we just havent implemented the logic to open up arrays and objects to expand their templated variables. Is that right?

Both are useful! This issue actually had both of those, an object key and also array values that were templated.

If thats true I'll leave it to you all to decide if object support is in addition to https://github.com/meltano/meltano/issues/3171 or if its big enough to be a separate issue.

**Comment by WillDaSilva** [comment]
Still relevant. It might be covered by #3171, but I'll play it safe and keep this issue open until we're sure of that (e.g. by verifying that in fixing #3171 we also fix this).
